### ISSUE TITLE
Status page refresh time

### ISSUE THREAD

**Issue by MuriloFuza** [louislam/uptime-kuma#5453]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

Would it be possible for me to change the status pages to update every 1 minute instead of 5 minutes?

### 🐻 Uptime-Kuma Version

1.23.15

### 💻 Operating System and Arch

alpine

### 🌐 Browser

Firefox

### 🖥️ Deployment Environment

- Runtime: 
- Database: 
- Filesystem used to store the database on: 
- number of monitors:

**Comment by homelab-alpha** [comment]
@MuriloFuza, this is not possible in version 1.23.15. However, it is available in version 2.0 BETA. You can check the [How to migrate from Version 1 to Version 2](https://github.com/louislam/uptime-kuma/wiki/Migration-From-v1-To-v2) for more details. Make sure to back everything up properly.

In my opinion, it's better to wait for version 2.1 BETA; I also advise against using the BETA in production.

**Comment by MuriloFuza** [comment]
Got it, I'll take a look and see the predictions. Thanks!

**Comment by gaby** [comment]
@homelab-alpha Not possible to add this as an ENV in the meantime?

I have a status page to monitor internet uptime, and the 5 minute refresh is too long.

The value is hard coded here: https://github.com/louislam/uptime-kuma/blob/master/server/socket-handlers/status-page-socket-handler.js#L288

**Comment by gaby** [comment]
Example of fixing this with a global ENV:

```javascript
// Read auto refresh interval from ENV variable with a default fallback to 300
const autoRefreshInterval = process.env.AUTO_REFRESH_INTERVAL
  ? parseInt(process.env.AUTO_REFRESH_INTERVAL, 10)
  : 300;

// ...

let statusPage = R.dispense("status_page");
statusPage.slug = slug;
statusPage.title = title;
statusPage.theme = "auto";
statusPage.icon = "";
statusPage.autoRefreshInterval = autoRefreshInterval;  // Use env var value here
await R.store(statusPage);
```

**Comment by homelab-alpha** [comment]
Hi @Gaby, you can adjust the value of the `Refresh Interval` on your `status page`. Go to your `status page`, click on `Edit Status Page`, and on the left side of the web page, you'll find the option `Refresh Interval`. This allows you to adjust the refresh rate of the `status page`. The value you enter is specified in seconds.

**TIP:** Do not set the value lower than 60 seconds. This prevents your hosting machine from becoming overloaded, especially if your `status page` is accessible online.

The default value for the automatic refresh interval is `statusPage.autoRefreshInterval = 300;`. In Uptime Kuma 2.0, this value is no longer hardcoded and can now be adjusted as described above.

I see no added value in making the `autoRefreshInterval` a Docker Compose ENV variable; it is already difficult enough for people to create and maintain a Compose file. Additionally, the `ENV` is not intended for this type of setting.

**Example where an ENV variable is useful for global Docker Compose settings:**

A typical use of an ENV variable in Docker Compose is for **global server settings** such as user ID, group ID, and timezone. These settings are often applied to multiple containers and need to be consistent at the server level. For example:

```yaml
environment:
  PUID: "1000"
  PGID: "1000"
  TZ: Europe/Amsterdam
```

In this case, these ENV variables ensure that:
- **PUID** and **PGID** set the correct user and group permissions for the container, which is important for access and file management.
- **TZ** sets the timezone for the container, ensuring it always uses the correct time without requiring you to modify the container code.

This type of setting is ideal for ENV variables because they are server-wide configurations that can easily be adjusted via the Docker Compose file without needing to modify the application code. For example, you can use the same settings for multiple containers, simplifying the configuration.

**Comment by homelab-alpha** [comment]
@Gaby, you can upgrade to Uptime Kuma `2.0.0-beta.1`. There is a small issue with deleting tags in the user interface, but this has been fixed in `2.0.0-beta.2`. Otherwise, the version is stable.

Feel free to give it a try.

Don't forget to back up! Back up! Back up! It may sound like overkill, but trust me, it's always better to be well-prepared! 😆

You can refer to the documentation on how to [migrate from v1 to v2](https://github.com/louislam/uptime-kuma/wiki/Migration-From-v1-To-v2).

```docker-compose.yml
image: louislam/uptime-kuma:2.0.0-beta.1
```

Just like you, I am also a community member of Uptime Kuma and have no influence on how and when a new beta or stable release is issued. When I suggested waiting for 2.1, I was convinced the beta releases would come much faster. You can follow the [Uptime Kuma milestones](https://github.com/louislam/uptime-kuma/milestones) here.
